SH4RE/SH4RE/Home/HomeView.swift

```diff
@@ -8,16 +8,138 @@
+                            guard searchModel.searchQuery.isEmpty == false else{ return }
+                            searchModel.searchReady = true
+                            tabSelection = 2
+                        }
+                    ScrollView {
+                        //Body
+                        VStack(alignment: .leading){
+                            //Recent Searches
+                            VStack (alignment: .leading) {
+                                Text("Recent Searches")
+                                    .font(.title2.bold())
+                                HStack(alignment: .center){
+                                    ForEach (searchModel.recentSearchQueries, id: \.self) { query in
+                                        Button(action: {
+                                            searchModel.searchQuery = query
+                                            searchModel.searchReady = true
+                                            tabSelection = 2
+                                        }) {
+                                            RecentSearchCard(searchText: query)
+                                            // match everything but the last
+                                            if query != searchModel.recentSearchQueries.last {
+                                                Spacer()
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            //Recent Posts
+                            VStack (alignment: .leading) {
+                                Text("Recent Posts")
+                                    .font(.title2.bold())
+                                HStack(){
+                                    ProductCard(listing: test_listing, image: UIImage(named: "ProfilePhotoPlaceholder")!)
+                                    Spacer()
+                                    ProductCard(listing: test_listing, image: UIImage(named: "ProfilePhotoPlaceholder")!)
+                                }
+                            }
+                            //Categories
+                            VStack (alignment: .leading) {
+                                Text("Categories")
+                                    .font(.title2.bold())
+                                VStack(spacing: 15){
+                                    ForEach(categories) {category in
+                                        CategoryCardView(tabSelection: $tabSelection, searchModel: searchModel, category: category)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    .scrollIndicators(.hidden)
+                }
+                .padding(.horizontal)
             }
         }
     }
 }
+
+
+struct RecentSearchCard: View {
```

Comment:
could this become its own component? similar to product card. up 2 u

Reply:
i figure no need, I broke it up so that it was easier to read, but I don't think putting it in its own file necessarily is much better. It will only be used in this one location and that's all, unlike product card which is used in a couple different places.